src/single-spa-react.js

```diff
@@ -207,30 +191,19 @@ function atLeastReact16(React) {
 function reactDomRender({ opts, elementToRender, domElement }) {
   const renderType =
     typeof opts.renderType === "function" ? opts.renderType() : opts.renderType;
-  if (
-    [
-      "createRoot",
-      "unstable_createRoot",
-      "createBlockingRoot",
-      "unstable_createBlockingRoot",
-    ].indexOf(renderType) >= 0
-  ) {
-    const root = opts.ReactDOM[renderType](domElement);
-    root.render(elementToRender);
-    return root;
-  }
-
-  if (renderType === "hydrate") {
-    opts.ReactDOM.hydrate(elementToRender, domElement);
-  } else {
-    // default to this if 'renderType' is null or doesn't match the other options
-    opts.ReactDOM.render(elementToRender, domElement);
+  switch (renderType) {
+    case "createRoot": {
+      const root = opts.ReactDOMClient.createRoot(domElement);
+      root.render(elementToRender);
+      return root;
+    }
+    case "hydrateRoot": {
+      const root = opts.ReactDOMClient.hydrateRoot(domElement, elementToRender);
```

Comment:
The migration docs made this specific callout:

> // Unlike with createRoot, you don't need a separate root.render() call here.

See also: https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis